### DISCUSSION TITLE
Manifold - General confusion about TriRef original ID

### DISCUSSION THREAD

**Discussion by yetigit** [elalish/manifold#399] [Q&A]
these are the values I get (click to enlarge)
![devenv_6IJSnAhxYA](https://user-images.githubusercontent.com/7016200/229376067-e03e5161-48d4-474c-b2e2-27a157237829.png)

## Here is what I expected  :

original Id's would be a bag of 0 and 1; 0 for the first sphere , 1 for the second sphere. 

also in this case all the values are "2" which means I can't know which mesh had the triangle.

**Comment by elalish** [comment]
Well, first question, have you checked the `originalID`s of your input meshes with `sphere.OriginalID()`? Second, how/why are you accessing the private members of `Manifold`? You get the original IDs from `GetMeshGLI()`.

**Comment by yetigit** [comment]
> Well, first question, have you checked the `originalID`s of your input meshes with `sphere.OriginalID()`? Second, how/why are you accessing the private members of `Manifold`? You get the original IDs from `GetMeshGLI()`.

@elalish 
i did check the original id  you can see it at the top of my debugger variable watch, for the 2 spheres it's 2

reason i can / i am looking privates ==> i exposed some things /  trying to understand the lib by looking at the data, also easier to use the interactive debugger this way.

**Comment by elalish** [comment]
Ah, missed those lines, thank you! Okay, I can explain what you're seeing now. So, you only constructed on object originally (the sphere) and then copied it, so their original ID is the same (2) as you noticed in the output. This is intentional: imagine you bring in multiple copies of the same textured object - likely you'll want to reapply its materials the same way on all copies. You can look at `runIndex` and `runTransform` in `MeshGL` to see which sets of triangles with the same `originalID` belong to different copies if you need to differentiate. There's also `AsOriginal()`, which will assign a new originalID to a given copy.
